addons/website/static/src/js/content/menu.js

```diff
@@ -660,6 +657,130 @@ publicWidget.registry.hoverableDropdown = animations.Animation.extend({
+        // Note: The XML templates have been modified to add them only once in
+        // the DOM. This code is needed to fix the issues for databases created
+        // before the templates modifications.
+        if (this.el.querySelectorAll(".o_mega_menu").length === 2 * this.desktopMegaMenuToggleEls.length) {
+            this.desktopMegaMenuToggleEls.forEach((megaMenuToggleEl, i) => {
+                const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+                if (hasMegaMenu) {
+                    const mobileMegaMenuEl = this.mobileMegaMenuToggleEls[i].parentElement.querySelector(".o_mega_menu");
+                    if (mobileMegaMenuEl) {
+                        mobileMegaMenuEl.remove();
+                    }
+                }
+            });
+        }
+
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * If the mega menu dropdown on which we are clicking/hovering does not have
+     * a mega menu (i.e. it is in the other navbar), brings the corresponding
+     * mega menu into it.
+     *
+     * @private
+     * @param {Element} megaMenuToggleEl the clicked/hovered mega menu dropdown
+     */
+    _moveMegaMenu(megaMenuToggleEl) {
+        const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+        if (hasMegaMenu) {
+            return;
+        }
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerUnactive("moveMegaMenu");
+        let megaMenuEl, previousDropdownEl;
+        const megaMenuId = megaMenuToggleEl.querySelector("span").dataset.oeId;
+        // If the `data-oe-id` attribute is there, we directly use it to find
+        // the corresponding mega menu (to be more efficient). Otherwise, we use
+        // the menus order.
+        if (megaMenuId) {
+            megaMenuEl = this.el.querySelector(`.o_mega_menu[data-oe-id="${megaMenuId}"]`);
+            previousDropdownEl = megaMenuEl.parentElement.querySelector(".o_mega_menu_toggle");
+        } else {
+            const isMobileNavbar = !!megaMenuToggleEl.closest(".navbar.o_header_mobile");
+            const currentNavbarToggleEls = isMobileNavbar ? this.mobileMegaMenuToggleEls : this.desktopMegaMenuToggleEls;
+            const otherNavbarToggleEls = isMobileNavbar ? this.desktopMegaMenuToggleEls : this.mobileMegaMenuToggleEls;
+            const megaMenuToggleIndex = currentNavbarToggleEls.indexOf(megaMenuToggleEl);
+            previousDropdownEl = otherNavbarToggleEls[megaMenuToggleIndex];
+            megaMenuEl = previousDropdownEl.parentElement.querySelector(".o_mega_menu");
+        }
+        // Hiding the previous dropdown before moving the mega menu.
+        Dropdown.getOrCreateInstance(previousDropdownEl).hide();
```

Comment:
Because if I move it before before hiding it, if it was somehow open somewhere else, then the mega menu has the `show` class while the dropdown does not have it yet (and the previous one still have it) and so the mechanism of hiding/showing gets broken (because the dropdowns and their menus are not synchronized) 

To put it simply, I hide it to be in a consistent state when moving the mega menu :slightly_smiling_face: 
I will add this as a comment :+1:

Reply:
For the same reason, I do it on `mouseDown` because the bootstrap hide/show event are fired on `click`, and I need to move the mega menus before bootstrap manages the event, otherwise the dropdowns are not are not in a correct state
For example, if I don't hide it manually, bootstrap will do it, but since the mega menu has been moved, there is a traceback because it cannot find the "popper", so the dropdown-menu, because it has been moved away

So I need to manage manually so bootstrap never sees that some dropdown are not "correct"

That's why I changed a foreach somewhere to only iterate on the shown dropdowns, because it would fail otherwise for the dropdowns whose mega menu has been moved away :slightly_smiling_face: